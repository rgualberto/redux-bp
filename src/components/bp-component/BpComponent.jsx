import './bp.scss';
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {toggleText} from './bpReducer';

export class BpComponent extends Component {
  toggle() {
    this.props.toggleText();
  }

  render() {
    const {
      buttonText,
      headingTitle,
      subHeading,
      titleToggleText,
      toggleState
    } = this.props;
    const title = toggleState ? headingTitle : titleToggleText;

    return (
      <div className="bp">
        <h1 className="bp__heading">{title}</h1>
        <p className="bp__subheading">{subHeading}</p>
        <button
          className="bp__text-toggle"
          onClick={this.toggle.bind(this)}
        >
          {buttonText}
        </button>
      </div>
    );
  }
}

BpComponent.propTypes = {
  buttonText: PropTypes.string,
  headingTitle: PropTypes.string,
  subHeading: PropTypes.string,
  titleToggleText: PropTypes.string,
  toggleState: PropTypes.bool,
  toggleText: PropTypes.func
};

function mapStateToProps(state) {
  return {
    toggleState: state.bpReducer.toggleState
  };
}

function mapDispatchToProps(dispatch) {
  return {
    toggleText: bindActionCreators(toggleText, dispatch),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BpComponent);
