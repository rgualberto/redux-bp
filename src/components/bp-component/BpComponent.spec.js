import React from 'react';
import {BpComponent} from './BpComponent.jsx';
import renderer from 'react-test-renderer';

describe('Boilerplate Component', () => {
  it('renders initial message', () => {
    const component = renderer.create(
      <BpComponent
        buttonText={"Toggle Title"}
        headingTitle={"Boilerplate"}
        subHeading={"Start with something rather than nothing!"}
        titleToggleText={"Base Dish"}
      />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
