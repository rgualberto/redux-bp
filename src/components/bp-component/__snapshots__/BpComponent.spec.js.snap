// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boilerplate Component renders initial message 1`] = `
<div
  className="bp"
>
  <h1
    className="bp__heading"
  >
    Base Dish
  </h1>
  <p
    className="bp__subheading"
  >
    Start with something rather than nothing!
  </p>
  <button
    className="bp__text-toggle"
    onClick={[Function]}
  >
    Toggle Title
  </button>
</div>
`;
